bot/extensions/vote_cog.py

```diff
@@ -0,0 +1,315 @@
+import random
+import asyncio
+
+
+# Problem: The modal doesn't allow more than 5 input fields (In the poll I assume there could be more)
+# Problem: Cannot make the bot output one modal after another (I can send 1 modal but after it says interaction expired or something).
+# When I try using webhooks it gives me 'Unknown Webhook' error
+# TODO(LIMITATION): Try to find a workaround with models. Or just accept user input: https://stackoverflow.com/questions/64949395/how-can-i-make-my-discord-bot-respond-with-a-follow-up-message
+
+
+#class TitleBuilder(ui.Modal):
+# 	def __init__(self):
+# 		super().__init__(title='Title')
+# 		self.input = ui.TextInput(label='Text')
+# 		self.add_item(self.input)
+
+# 	async def on_submit(self, interaction: discord.Interaction):
+# 		await interaction.response.send_message(f'Text: {self.input}', ephemeral=True)
+
+# class OptionBuilder(ui.Modal):
+# 	def __init__(self, option: int):
+# 		super().__init__(title=f'Option {option}')
+# 		self.option = ui.TextInput(label=f"Option {option}")
+# 		self.add_item(self.option)
+
+# 	async def on_submit(self, interaction: discord.Interaction):
+# 		await interaction.response.send_message(f'Text: {self.option}', ephemeral=True)
+
+class PollView(View):
+	def __init__(self, emojis: list[str], possible_emojis_size: int, ref_embed: Embed) -> None:
+		super().__init__()
+		self.buttons = []
+		for emoji_index in range(possible_emojis_size):
+			button = VoteButton(emojis[emoji_index])
+			button.set_embed_reference(ref_embed)
+			self.add_item(button)
+			self.buttons.append(button)
+
+	def disable_buttons(self) -> None:
+		for button in self.buttons:
+			button.disabled = True
+
+
+class PollEmbed(Embed):
+	def __init__(self, *, options: list[str], emojis: list[str], counter: dict, title: str) -> None:
+		super().__init__()
+		# User: vote_emoji
+		self._voted_users = {}
+		self._poll_options = options
+		self._poll_emojis = emojis
+		self._poll_counter = counter
+		self._poll_title = title
+		self._timer_label = None
+
+	def increment_counter(self, emoji: str) -> None:
+		self._poll_counter[emoji] += 1
+
+	def decrement_counter(self, emoji: str) -> None:
+		self._poll_counter[emoji] -= 1
+
+	def set_user(self, user: discord.Member, emoji: str) -> None:
+		self._voted_users[user] = emoji
+
+	def get_user_emoji(self, user: discord.Member) -> str:
+		return self._voted_users[user]
+
+	def user_voted(self, user: discord.Member) -> bool:
+		if user in self._voted_users:
+			return True
+		return False
+
+	def set_timer_label(self, label: str) -> None:
+		self._timer_label = label
+
+	async def get_and_print_winner(self, ctx: Context) -> None:
+		highest = 0
+		win_emoji = ''
+		for emoji, count in self._poll_counter.items():
+			if count > highest:
+				highest = count
+				win_emoji = emoji
+
+		if highest > 0 and win_emoji:
+			await ctx.channel.send(f'{win_emoji} ***option has won!***')
+		else:
+			await ctx.channel.send('No one voted.')
+
+
+
+	def build(self) -> None:
```

Comment:
I might be annoying with that, but I am very pro-function documentation. I would recommend adding a docstring after each function that quickly describes what the function is doing. It could be a simple sentence that summarises what the function is made for, but I think it is good practice to do so.

Reply:
No that's great, I will do that.